civis/tests/test_io.py

```diff
@@ -398,6 +398,147 @@ def test_civis_file_to_table_table_doesnt_exist(
             **expected_kwargs
         )
 
+    @pytest.mark.civis_file_to_table
+    @mock.patch('civis.io._tables._process_cleaning_results')
+    @mock.patch('civis.io._tables._run_cleaning')
+    def test_civis_file_to_table_table_exists_all_sql_types_missing(
+            self,
+            m_run_cleaning,
+            m_process_cleaning_results,
+            _m_get_api_spec
+    ):
+        table = "scratch.api_client_test_fixture"
+        database = 'redshift-general'
+        mock_file_id = 1234
+        mock_cleaned_file_id = 1235
+        mock_import_id = 8675309
+
+        self.mock_client.imports.post_files_csv.return_value\
+            .id = mock_import_id
+        self.mock_client.get_database_id.return_value = 42
+        self.mock_client.default_credential = 713
+
+        table_columns = [{'name': 'a', 'sql_type': ''},
+                         {'name': 'b', 'sql_type': ''}]
+        m_process_cleaning_results.return_value = (
+            [mock_cleaned_file_id],
+            True,  # headers
+            'gzip',  # compression
+            'comma',  # delimiter
+            table_columns  # table_columns
```

Comment:
In this test, the `table_columns` provided as the parameter (to `civis.io.civis_file_to_table`) should be different than what is returned by `m_process_cleaning_results` or our cleaning job. This is because that job is detecting sql types since sql types were not provided and the table does not exist or is being dropped. So I would expect this `table_columns` to return valid `sql_type` entries.

Reply:
yeah I was wondering that! Looking at the existing tests it looked like they were passing in the same table_columns so I was confused about that, but I copied them. I'll take a closer look!

Reply:
done

Reply:
Yeah that makes sense!